api/pb/android/threads/src/main/java/io/textile/threads/Client.java

```diff
@@ -0,0 +1,155 @@
+package io.textile.threads;
+
+import java.io.ByteArrayOutputStream;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import android.arch.lifecycle.LifecycleObserver;
+
+import com.google.protobuf.ByteString;
+
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+
+import java.util.Iterator;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import io.grpc.stub.StreamObserver;
+import io.textile.threads_grpc.*;
+
+
+class ClientException extends Exception
+{
+    public ClientException(String message)
+    {
+        super(message);
+    }
+}
+
+/**
+ * Provides top level access to the Textile API
+ */
+public class Client implements LifecycleObserver {
```

Comment:
the bulk of the existing api implementations. focused on the easy to do sync methods until all of it is working well